playbooks/roles/deploy-osh/tasks/enroll-caasp-nodes-for-osh.yml

```diff
@@ -71,14 +76,12 @@
   with_items:
     - 'openstack-helm-node-class=primary'
     - 'ceph-mon=enabled'
-  when:
-    - "(master_node | default('')) | length > 0"
+  when: _primary_node is defined
 
 # TODO(evrardjp): Move this to check the return code and task content
 - name: Remove taints when only 3 nodes and ha is required
-  command: "kubectl taint nodes {{ item }} node-role.kubernetes.io/master:NoSchedule-"
-  with_items: "{{ master_node }}"
+  command: "kubectl taint nodes {{ _master_node.stdout }} node-role.kubernetes.io/master:NoSchedule-"
   when:
-    - "(master_node | default('')) | length > 0"
+    - _master_node is defined
```

Comment:
nit: It will always be defined, else it won't run this task because of a failure :) 
You can remove this.